dataflux_pytorch/benchmark/checkpointing/multinode/train.py

```diff
@@ -0,0 +1,294 @@
+                        optimizer_key=optim_key,
+                        storage_reader=self.reader,
+                    )
+                    flattened_osd = FSDP.optim_state_dict_to_load(
+                        optim_state_dict=optim_state[optim_key],
+                        model=self.model,
+                        optim=optim,
+                    )
+                    optim.load_state_dict(flattened_osd)
+
+        # Load metadata (anything not a module or optimizer)
+        new_path = path / _METADATA_FILENAME
+        metadata = None
+        with self.reader.fs.create_stream(path=new_path, mode='rb') as metadata_file:
+            metadata = torch.load(metadata_file)
+        return metadata
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--project", type=str)
+    parser.add_argument("--ckpt-dir-path", type=str)
+    parser.add_argument("--ckpt_restore_path", type=str)
+    parser.add_argument("--save-only-latest",
+                        action="store_true",
+                        default=False)
+    parser.add_argument("--no-dataflux-ckpt",
+                        action="store_true",
+                        default=False)
+    parser.add_argument("--layers", type=int, default=100)
+    parser.add_argument("--steps", type=int, default=5)
+    parser.add_argument("--enable-multipart",
+                        action="store_true",
+                        default=False)
+    parser.add_argument("--min-epochs-save", type=int, default=4)
+    parser.add_argument("--max-epochs-save", type=int, default=5)
+    parser.add_argument("--max-steps-save", type=int, default=5)
+    parser.add_argument("--min-epochs-restore", type=int, default=4)
+    parser.add_argument("--max-epochs-restore", type=int, default=5)
+    parser.add_argument("--max-steps-restore", type=int, default=5)
+    return parser.parse_args()
+
+
+def configure_master_addr():
+    """Get coordinator IP Address with retries"""
+    coordinator_address = ""
+    coordinator_ip_address = ""
+    if os.environ.get("COORDINATOR_ADDRESS") is not None:
+        coordinator_address = os.environ.get("COORDINATOR_ADDRESS")
+        coordinator_found = False
+        lookup_attempt = 1
+        max_coordinator_lookups = 50
+        while not coordinator_found and lookup_attempt <= max_coordinator_lookups:
+            try:
+                coordinator_ip_address = socket.gethostbyname(
+                    coordinator_address)
+                coordinator_found = True
+            except socket.gaierror:
+                print(
+                    f"Failed to recognize coordinator address {coordinator_address} on"
+                    f" attempt {lookup_attempt}, retrying...")
+                lookup_attempt += 1
+                time.sleep(5)
+    print(f"Coordinator IP address: {coordinator_ip_address}")
+    os.environ["MASTER_ADDR"] = str(coordinator_ip_address)
+
+
+def init_processes():
```

Comment:
Code in this function and maybe some other places seems to still be using environment variables instead of flags. Is that intended? If so please document the required environment variables in the readme.

Reply:
I have reverted the code to use environment variables instead of flags, since some work will be needed to make flags code work with GKE. I dont see a point in blocking benchmarking on it right now. 

Also documented the environment variables.